### PR TITLE
replace checks for platform with checks for platform_family

### PR DESCRIPTION
**Pull request by jsuchome**

### CODE REVIEW

File: chef/cookbooks/bluepill/test/cookbooks/bluepill_test/recipes/default.rb [hunk 1]
```diff
@@ -2,12 +2,7 @@
 
 # Boring
 package "nc" do
-  package_name case node["platform_family"]
-               when "debian"
-                 "netcat"
-               else
-                 "nc"
-               end
+  package_name = "netcat" if node[:platform_family] == "debian"
```
Comment (jsuchome) [inline chef/cookbooks/bluepill/test/cookbooks/bluepill_test/recipes/default.rb]: This doesn't make sense, does it?

**Comment by aplanas** [comment]
* chef/cookbooks/uwsgi/resources/default.rb L19
* chef/cookbooks/bind9/recipes/default.rb L232, L250
* chef/cookbooks/apache2/templates/default/mods/authopenid.load.erb L1
* chef/cookbooks/apache2/templates/default/mods/ssl.conf.erb L42, L52
* chef/cookbooks/apache2/templates/default/mods/fcgid.conf.erb L6
* chef/cookbooks/apache2/templates/default/mods/mime.conf.erb L7
* chef/cookbooks/apache2/templates/default/apache2.conf.erb L11, L13, L23, L25, L27
* chef/cookbooks/apache2/templates/default/default-site.erb L57
* chef/cookbooks/apache2/attributes/default.rb L20-L34
* chef/cookbooks/apache2/recipes/default.rb L49
* chef/cookbooks/network/recipes/default.rb L29
* chef/cookbooks/provisioner/libraries/repositories.rb L85
* chef/cookbooks/provisioner/recipes/setup_base_images.rb L117
  `if node[:platform] == "suse" && node[:platform_version].to_f < 12.0`
* chef/cookbooks/provisioner/recipes/base.rb L165, L189, L234
  `if node[:platform] == "suse" && node[:platform_version].to_f < 12.0`
* chef/cookbooks/crowbar/templates/default/rainbows.cfg.erb L10
  `<% if node[:platform]=="suse" -%>`
* chef/cookbooks/crowbar/recipes/default.rb L278
  `if node[:platform] == "suse" && node[:platform_version].to_f < 12.0`
* chef/cookbooks/openldap/templates/default/slapd.conf.erb L30
* chef/cookbooks/openldap/recipes/server.rb L21, L62
* chef/cookbooks/ntp/recipes/default.rb L93
* chef/cookbooks/logging/recipes/server.rb L45
* chef/cookbooks/logging/recipes/common.rb L33
  `only_if { node[:platform] == "suse" && node[:platform_version].to_f < 12.0`
* chef/cookbooks/logging/recipes/common.rb L38, L39
* chef/cookbooks/runit/recipes/default.rb L64
* chef/cookbooks/openssh/recipes/default.rb L47
* chef/cookbooks/ruby/definitions/ruby_symlinks.rb L22
* chef/cookbooks/dhcp/recipes/default.rb L156
* crowbar_framework/barclamp_model/chef/cookbooks/==BC-MODEL==/definitions/==BC-MODEL==_service.rb L6
* crowbar_framework/lib/remote_node.rb L127
* crowbar_framework/app/assets/javascripts/jquery/nodeList.js: This javascript code refers to platform too

**Comment by jsuchome** [comment]
@aplanas I think most of them are correct, sometimes it makes sense to check platform and not platform version, esp. together with version check

**Comment by nkrinner** [comment]
I think we can move to platform_family in those cases:

https://github.com/jsuchome/crowbar-core/blob/ec3fd812ec4a26279f4f895d18af1841f0b10c00/chef/cookbooks/repos/recipes/default.rb#L18 https://github.com/jsuchome/crowbar-core/blob/ec3fd812ec4a26279f4f895d18af1841f0b10c00/chef/cookbooks/provisioner/libraries/repositories.rb#L85
https://github.com/jsuchome/crowbar-core/blob/ec3fd812ec4a26279f4f895d18af1841f0b10c00/chef/cookbooks/provisioner/recipes/setup_base_images.rb#L104 (also a part can be dropped, we don't support a version < 12.0)
https://github.com/jsuchome/crowbar-core/blob/ec3fd812ec4a26279f4f895d18af1841f0b10c00/chef/cookbooks/provisioner/recipes/setup_base_images.rb#L311
https://github.com/jsuchome/crowbar-core/blob/ec3fd812ec4a26279f4f895d18af1841f0b10c00/chef/cookbooks/provisioner/recipes/base.rb#L165 (we don't support a version <12.0, similar same file line 189, line 234)
https://github.com/jsuchome/crowbar-core/blob/ec3fd812ec4a26279f4f895d18af1841f0b10c00/chef/cookbooks/nfs-server/recipes/default.rb#L45

There is a number of other cases where we check if platform_version is <12.0. but I stopped listing them here. It is my understanding, that we do not need those passages anymore and they should be dropped.

**Comment by vuntz** [comment]
@jsuchome see https://github.com/jsuchome/crowbar-core/pull/1 for some suggestions.

**Comment by jsuchome** [comment]
@nkrinner those provisioner bits imho deserve check for platform (= SLES)

**Comment by jsuchome** [comment]
And I'm not sure if we should just drop all checks for 12.0 and treat SLE12 as only supported. Shouldn't it be target for different PR?

**Comment by vuntz** [comment]
> And I'm not sure if we should just drop all checks for 12.0 and treat SLE12 as only supported.

Please don't. We might add SLE11SP4 later on.

**Comment by vuntz** [comment]
+1; I think there might be more to do in the rails app, but that's a good start that I feel we could already merge.

**Comment by aplanas** [comment]
+1 Lets wait for the gating

**Comment by rsalevsky** [comment]
They are still some hound issues open which can be fixed easily.

**Comment by vuntz** [comment]
This needs to be rebased :/

File: chef/cookbooks/bind9/recipes/default.rb [hunk 2]
```diff
@@ -246,6 +248,17 @@ def make_zone(zone)
...
+service "bind9" do
+  case node[:platform_family]
+  when "rhel", "suse"
+    service_name "named"
+  end
+  supports restart: true, status: true, reload: true
+  running true
+  enabled true
+  action :enable
+end
```
Comment (vuntz) [inline chef/cookbooks/bind9/recipes/default.rb]: Wrong rebase here: this service definition actually got moved, a few lines later, so you should not readd it, but update the moved one.

Comment (jsuchome) [inline chef/cookbooks/bind9/recipes/default.rb]: Too many rebases already :-(
I'm glad someone reads it after me, thanks

File: chef/cookbooks/apache2/recipes/default.rb [hunk 3]
```diff
@@ -146,13 +147,11 @@
...
-    case node[:platform]
-    when "centos","redhat","fedora","arch"
+    case node[:platform_family]
+    when "rhel", "fedora", "arch"
       path "#{node[:apache][:dir]}/conf/httpd.conf"
-    when "debian","ubuntu"
+    when "debian"
       path "#{node[:apache][:dir]}/apache2.conf"
-    when "suse"
-      path "#{node[:apache][:dir]}/httpd.conf"
```
Comment (vuntz) [inline chef/cookbooks/apache2/recipes/default.rb]: I guess the `suse` bit should not be dropped?

**Comment by vuntz** [comment]
+1

**Comment by aplanas** [comment]
+1

**Pull request merged**